executor/executor.go

```diff
@@ -1111,6 +1118,107 @@ func (e *AggregationExec) innerNext() (ret bool, err error) {
 	return true, nil
 }
 
+// StreamAggExec deals with all the aggregate functions.
+// It is built from Aggregate Plan. When Next() is called, it reads all the data from Src and updates all the items in AggFuncs.
+type StreamAggExec struct {
+	Src          Executor
+	schema       expression.Schema
+	ResultFields []*ast.ResultField
+	executed     bool
+	hasData      bool
+	mode         plan.AggregationType
+	ctx          context.Context
+	AggFuncs     []expression.AggregationFunction
+	GroupByItems []expression.Expression
+}
+
+// Close implements Executor Close interface.
+func (e *StreamAggExec) Close() error {
+	e.executed = false
```

Comment:
`executed` need three kinds of state: not executed, executed, closed
we can't distinguish between not executed and closed if set `e.executed = false` here...